tests/test_mig_install_generateconfs.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import print_function
+import importlib
+import os
+import sys
+
+from tests.support import MIG_BASE, MigTestCase, testmain, cleanpath
+
+
+def _import_generateconfs():
+    sys.path.append(os.path.join(MIG_BASE, 'mig/install'))
+    mod = importlib.import_module('generateconfs')
+    sys.path.pop(-1)
+    return mod
+
+
+# workaround for generatconfs being placed witin a non-module directory
+generateconfs = _import_generateconfs()
+main = generateconfs.main
+
+
+def create_fake_generate_confs(return_dict=None):
+    def _generate_confs(*args, **kwargs):
+        return return_dict if return_dict else {}
+    return _generate_confs
+
+
+class MigInstallGeneateconfs__main(MigTestCase):
```

Comment:
I'll fix the 'Geneate' typo during merge and add missing doc strings this time. I know it's boring and that it can slip, but please try to get in the habit of writing a line or two for anything non-trivial. I know from experience that they are helpful when revisiting weeks or months later.

Reply:
I will re-enforce doing this up-front - I had been making a point of this but have slipped a little. On my radar again.